graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/migration/CreateIndexQuery.java

```diff
@@ -55,6 +61,18 @@ public String getDescription() {
         "Create index %s on %s.%s", index.name(), table.name(), index.column().name());
   }
 
+  @Override
+  public void authorize(AuthorizationService authorizationService, AuthenticationSubject subject)
+      throws UnauthorizedException {
+    authorizationService.authorizeSchemaWrite(
+        subject,
+        table.keyspace(),
+        table.name(),
+        Scope.ALTER,
+        SourceAPI.GRAPHQL,
+        ResourceKind.TABLE);
```

Comment:
Not sure about this one, so I used the rule that is implemented [in Cassandra](https://github.com/apache/cassandra/blob/trunk/src/java/org/apache/cassandra/cql3/statements/schema/CreateIndexStatement.java#L218): creating an index requires ALTER permission on the target table.

Reply:
`StargateQueryHandler` [uses](https://github.com/stargate/stargate/blob/1bfee7b1d080017c2496ee43e4225e6d9891d3cf/persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/StargateQueryHandler.java#L492) `CREATE` and `INDEX` for authorizing `CreateIndexStatement`.

Reply:
Good call, it didn't occur to me to look at `StargateQueryHandler`, but it makes sense to align the checks with it.